nav2_dynamic_motion_model/include/nav2_dynamic_motion_model/motion_model_interface.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef NAV2_DYNAMIC_MOTION_MODEL__MOTION_MODEL_INTERFACE_HPP_
+#define NAV2_DYNAMIC_MOTION_MODEL__MOTION_MODEL_INTERFACE_HPP_
+
+#include <memory>
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/pose_array.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <nav2_dynamic_msgs/msg/obstacle.hpp>
+#include <nav2_dynamic_msgs/msg/obstacle_array.hpp>
+
+namespace nav2_dynamic_motion_model
+{
+
+class MotionModelInterface
+{
+public:
+    using Ptr = std::shared_ptr<MotionModelInterface>;
+    using ConstPtr = std::shared_ptr<const MotionModelInterface>;
+
+    virtual ~MotionModelInterface() = default;
+
+    virtual geometry_msgs::msg::Pose predictObstaclePose(
+        const nav2_dynamic_msgs::msg::Obstacle &obstacle,
```

Comment:
Is the assumption that whatever is calling the `predict` methods on these objects is already going to find the `dt` w.r.t. the last obstacle message instance timestamp and the request time? 

I don't think it'll be so easy, given that the messages come into the utility function at some arbitrary rate from the obstacle detector/tracking pipeline. We may need to take messages from some non-fixed time ago and the current time to find thees values. 

It would also be good to consider the covariance for the output so we have some confidence in the output pose (i.e. `geometry_msgs::msg::PoseStampedWithCovariance`